Source/diablo.cpp

```diff
@@ -850,101 +850,57 @@ BOOL __fastcall LeftMouseDown(int a1)
 // 69BD04: using guessed type int questlog;
 // 6AA705: using guessed type char stextflag;
 
-BOOL __fastcall LeftMouseCmd(BOOL a1)
+BOOL __fastcall LeftMouseCmd(BOOL bShift)
 {
-	BOOLEAN v2; // zf
-	BOOL v3; // ecx
-	unsigned char v4; // dl
-	unsigned char v6; // dl
-	unsigned short v7; // ax
-	unsigned char v8; // dl
-	unsigned short v11; // [esp-4h] [ebp-Ch]
-
-	if ( leveltype )
-	{
-		v3 = abs(plr[myplr].WorldX - cursmx) < 2 && abs(plr[myplr].WorldY - cursmy) < 2;
-		if ( pcursitem != -1 && pcurs == 1 && !a1 )
-		{
-			NetSendCmdLocParam1(1u, (invflag == 0) + CMD_GOTOGETITEM, cursmx, cursmy, pcursitem);
-LABEL_47:
-			if ( pcursitem != -1 )
-				return 0;
-			v2 = pcursobj == -1;
-			goto LABEL_49;
-		}
-		if ( pcursobj != -1 && (!a1 || v3 && object[pcursobj]._oBreak == 1) )
-		{
-			NetSendCmdLocParam1(1u, (pcurs == 5) + CMD_OPOBJXY, cursmx, cursmy, pcursobj);
-			goto LABEL_46;
-		}
-		if ( plr[myplr]._pwtype == 1 )
-		{
-			if ( a1 )
-			{
-				v4 = CMD_RATTACKXY;
-LABEL_35:
-				NetSendCmdLoc(1u, v4, cursmx, cursmy);
-				goto LABEL_46;
-			}
-			if ( pcursmonst != -1 )
-			{
-				v11 = pcursmonst;
-				if ( !CanTalkToMonst(pcursmonst) )
-				{
-					v6 = CMD_RATTACKID;
-LABEL_40:
-					NetSendCmdParam1(1u, v6, v11);
-					goto LABEL_47;
+	BOOL bNear;
+
+	if(!leveltype) {
+		if(pcursitem != -1 && pcurs == CURSOR_HAND)
+			NetSendCmdLocParam1(1, invflag ? CMD_GOTOGETITEM : CMD_GOTOAGETITEM, cursmx, cursmy, pcursitem);
+		if(pcursmonst != -1)
+			NetSendCmdLocParam1(1, CMD_TALKXY, cursmx, cursmy, pcursmonst);
+		if(pcursitem == -1 && pcursmonst == -1 && pcursplr == -1)
+			return TRUE;
+	} else {
+		bNear = abs(plr[myplr].WorldX - cursmx) < 2 && abs(plr[myplr].WorldY - cursmy) < 2;
+		if(pcursitem != -1 && pcurs == CURSOR_HAND && !bShift) {
+			NetSendCmdLocParam1(pcurs, invflag ? CMD_GOTOGETITEM : CMD_GOTOAGETITEM, cursmx, cursmy, pcursitem);
+		} else if(pcursobj != -1 && (!bShift || bNear && object[pcursobj]._oBreak == 1)) {
+			NetSendCmdLocParam1(1, pcurs == CURSOR_DISARM ? CMD_DISARMXY : CMD_OPOBJXY, cursmx, cursmy, pcursobj);
+		} else if(plr[myplr]._pwtype == 1) {
+			if(bShift) {
+				NetSendCmdLoc(1, CMD_RATTACKXY, cursmx, cursmy);
+			} else if(pcursmonst != -1) {
+				if(CanTalkToMonst(pcursmonst)) {

```

Comment:
The case here seams to need to be flipped i order for it to produce bin exact code.
```suggestion
				if(!CanTalkToMonst(pcursmonst)) {
					NetSendCmdParam1(TRUE, CMD_RATTACKID, pcursmonst);
				} else {
					NetSendCmdParam1(TRUE, CMD_ATTACKID, pcursmonst);
				}
```

Reply:
The function submitted as is produced bin exact code for me using VC6 with SP5 PPC.

Reply:
odd, at least the difference is easy to spot and correct so probably lets not investigate it further for now. yours also look more logical.

Reply:
Looks like I had an old instillation of VC on one of my computeres, interesting that it swaps if-else cases in some cases.

Reply:
We should really put up a *de-facto* VC install somewhere, so we can ensure that everyone has the right version.

Reply:
Well we kinda have, but the problem was that I hadn't pulled the latest image from https://hub.docker.com/r/diasurgical/riivaaja/tags/ so I was still running an old one from before the latest tweaks to it. Once we switch to the C compiler we may have to update things again based on new findings.